src/plugins/merge/merge.js

```diff
@@ -0,0 +1,40 @@
+import _set from 'lodash/fp/set';
+
+import { convertPathToKeys, resolveReferences } from './referenceResolver';
+
+const mergeIncludedReferenceIntoEntity = (entity, included, reference, path) => {
+  const includedEntity = included[reference.type].find(includedEntry => includedEntry.id === reference.id);
+
+  return _set(path, { ...reference, ...includedEntity }, entity);
```

Comment:
Q: I am not that familiar with this `set` function from Lodash, but I looked it up and it seems like the arguments you pass to it, are not in the right order. 

The docs say it should be `(object, path, value)`, but it seems like you have `(path, value, object)`. Am I missing something here?

Reply:
This is the `lodash/fp` version of `set`, which returns a new object, as opposed to mutating it. For some reason the arguments are in a different order here, don't ask 🙄 

https://gist.github.com/jfmengels/6b973b69c491375117dc#_setpath-value-object

Reply:
aha, all good then :)

Reply:
> For some reason the arguments are in a different order here, don't ask

I think it's for currying reasons :)